breathecode/marketing/tests/urls/tests_lead_id.py

```diff
@@ -0,0 +1,266 @@
+            'language': model.form_entry[0].language,
+            'last_name': model.form_entry[0].last_name,
+            'lead_type': model.form_entry[0].lead_type,
+            'location': model.form_entry[0].location,
+            'storage_status': model.form_entry[0].storage_status,
+            'tags': model.form_entry[0].tags,
+            'utm_campaign': model.form_entry[0].utm_campaign,
+            'utm_medium': model.form_entry[0].utm_medium,
+            'utm_content': model.form_entry[0].utm_content,
+            'utm_source': model.form_entry[0].utm_source,
+            'utm_url': model.form_entry[0].utm_url,
+            'latitude': model.form_entry[0].latitude,
+            'longitude': model.form_entry[0].longitude,
+            'phone': model.form_entry[0].phone,
+            'user': model.form_entry[0].user,
+            'referral_key': model.form_entry[0].referral_key,
+            'state': model.form_entry[0].state,
+            'storage_status_text': model.form_entry[0].storage_status_text,
+            'street_address': model.form_entry[0].street_address,
+            'zip_code': model.form_entry[0].zip_code
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    """
+    🔽🔽🔽 Update lead with wrong id
+    """
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_update_lead_wrong_id(self):
+        """Test /lead/:id/ with data wrong id"""
+        self.headers(academy=1)
+        url = reverse_lazy('marketing:academy_lead_id', kwargs={'lead_id': 2})
+        model = self.generate_models(
+            authenticate=True,
+            profile_academy=True,
+            capability='crud_lead',
+            role='potato',
+            form_entry=True,
+        )
+
+        data = {'first_name': 'Juan'}
+        response = self.client.put(url, data, format='json')
+        json = response.json()
+
+        expected = {'detail': 'lead-not-found', 'status_code': 400}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 400)
+
+    """
+    🔽🔽🔽 Update lead
+    """
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_update_lead(self):
+        """Test /lead/:id/ with data wrong id"""
+        self.headers(academy=1)
+        url = reverse_lazy('marketing:academy_lead_id', kwargs={'lead_id': 1})
+        model = self.generate_models(
+            authenticate=True,
+            profile_academy=True,
+            capability='crud_lead',
+            role='potato',
+            form_entry=True,
+        )
+
+        data = {'id': 1, 'first_name': 'Juan Arango'}
```

Comment:
pass all the possible values can be changed, and the attrs needs be random value, use `self.bc.fake` it's a instance of faker